contrib/pg_tde/documentation/docs/replication.md

```diff
@@ -1,4 +1,152 @@
-# Replication
+# Streaming Replication with tde_heap
 
-<!-- To add information on Replication affecting pg_tde for customers and how it interacts with it
-Also, don't forget to add it to TOC>
\ No newline at end of file
+This section outlines how to set up PostgreSQL streaming replication when the `pg_tde` extension (specifically the `tde_heap` access method) is enabled on the primary server.
+
+The following steps assume:
+
+* You have configured a global key provider for both **primary** and **standby** (see [Configure Key Management (KMS)](global-key-provider-configuration/index.md)).
+* You have enabled `pg_tde` and you have setup at least one active key on the **primary**.
+* Both primary and standby run the **same** Percona PostgreSQL version.
+
+## 1. Configure the Primary
+
+### Configure postgresql.conf
+
+* Ensure you set the following replication settings in `postgresql.conf`:
+
+    ```ini
+    # Example of WAL and replication settings
+    wal_level            = replica
+    max_wal_senders      = 5
+    max_replication_slots = 1
+    wal_keep_size        = '1GB'
+    # Enable TDE in WAL pipeline
+    shared_preload_libraries = 'pg_tde' # Loads TDE hooks at server start.
+    ```
+
+    !!! note
+        Make sure you set `max_replication_slots` before creating any slot.
+
+* (Optional - **To review**) Create a physical slot to retain the encrypted WAL, set `max_replication_slots ≥ 1` and then:
+
+    ```sql
+    SELECT pg_create_physical_replication_slot('tde_slot');
+    ```
+
+* Ensure you have [configured the global key provider](global-key-provider-configuration/index.md).
+* Create the [principal key](functions#pg_tde_set_server_key_using_global_key_provider).
+* Enable [WAL encryption](wal-encryption). **Restart** PostgreSQL.
+* Ensure the extension is installed in each database:
+
+    ```sql
+    CREATE EXTENSION IF NOT EXISTS pg_tde;
+    ```
+
+* (Optional) If you want to block any unencrypted tables, you can [enforce table-level encryption](variables#pg_tde.enforce_encryption):
```

Comment:
This and some other information here - again I don't see how is this related to replication?